### PR TITLE
feat: add `Guild.edit_mfa_level`

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

see https://github.com/discord/discord-api-docs/commit/a54996ba563e88eebf213c96fbb3089bf3cdc7ae

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/guild.py [hunk 1]
```diff
@@ -185,9 +185,9 @@ class Guild(Hashable):
     description: Optional[:class:`str`]
         The guild's description.
     mfa_level: :class:`int`
-        Indicates the guild's two factor authorisation level. If this value is 0 then
-        the guild does not require 2FA for their administrative members. If the value is
-        1 then they do.
+        Indicates the guild's two-factor authentication level. If this value is 0 then
+        the guild does not require 2FA for their administrative members
+        to take moderation actions. If the value is 1 then it does.
```
Comment (onerandomusername) [inline disnake/guild.py:190]:
```suggestion
        to take moderation actions. If the value is 1 then the guild does.
```

File: disnake/guild.py [hunk 2]
```diff
@@ -3752,6 +3752,36 @@ def widget_image_url(self, style: WidgetStyle = WidgetStyle.shield) -> str:
...
+        ------
+        HTTPException
+            Editing the 2FA level failed.
+        ValueError
+            You are not the owner of the guild.
+        """
+        if self.owner_id != self._state.self_id:
+            raise ValueError("To edit the 2FA level, you must be the owner of the guild.")
+        # return value unused
+        await self._state.http.edit_mfa_level(self.id, mfa_level, reason=reason)
```
Comment (onerandomusername) [inline disnake/guild.py:3783]: we should catch the typecheck issue of mfa_level being a bool

Comment (shiftinv) [reply disnake/guild.py:3785]: 63507c4b437f775218216b09738764d55e2c9472

File: disnake/guild.py (hunk 1 above)
Comment (shiftinv) [reply disnake/guild.py]: good call, adjusted it slightly more: 8571578754d66c892c4c093787aef16b94251146

**Review by onerandomusername — APPROVED** [review]
looks good.

**Pull request merged**